main.cpp

```diff
@@ -46,20 +46,75 @@ Project 4: Part 4 / 9
  You will need to use Forward Declaration and out-of-class definitions to complete this.
  */
 
+#include "IntType.h"
+#include "FloatType.h"
+#include "DoubleType.h"
 
 
 struct Point
 {
+    Point(DoubleType& x_, DoubleType& y_);
+    Point(IntType& x_, IntType& y_);
+    Point(FloatType& x_, FloatType& y_);
+    Point(float x_, float y_) : x { x_} , y { y_} {}
+
     Point& multiply(float m)
     {
         x *= m;
         y *= m;
         return *this;
     }
+
+    Point& multiply(DoubleType& m)
+    {
+        x *= static_cast<float>(m);
+        y *= static_cast<float>(m);
+        return *this;
+    }
+
+    Point& multiply(IntType& m)
+    {
+        x *= m;
+        y *= m;
+        return *this;
+    }
+
+    Point& multiply(FloatType& m)
+    {
+        x *= m;
+        y *= m;
+        return *this;
+    }
+
+    void toString()
+    {
+        std::cout << "{ x: " << this->x << ", y: " << this->y << " }" << std::endl;
+    }
+
+
 private:
     float x{0}, y{0};
 };
 
+
+Point::Point(DoubleType& x_, DoubleType& y_)
+{
+    x = static_cast<float>(x_);
+    y = static_cast<float>(y_);
+}
+
+Point::Point(FloatType& x_, FloatType& y_)
+{
+    x = x_;
+    y = y_;
+}
+
+Point::Point(IntType& x_, IntType& y_)
+{
+    x = x_;
+    y = y_;
+}
```

Comment:
Call the constructor that takes primitives.  pass by const-ref, not by ref.  Never pass by ref, unless you need to modify the passed-in value and have those changes present after the function exits.
- [ ] revise